### DISCUSSION TITLE
Extending QR-code plugin with multi-vendor scanning/provisioning capability

### DISCUSSION THREAD

**Discussion by jbemmel** [netbox-community/netbox#11004] [Ideas]
Hi everyone,
Some of our customers use barcode scanners to track equipment going into their data centers, and we're considering to contribute a feature to support such a workflow to provision a Netbox instance.

At https://jbemmel.github.io/netbox-qrprovision/develop/read-qrcode.html you can get an impression of what this might look like, using an open source html5 QR code library. Initial idea is to add this as a feature for the existing QR plugin, see https://github.com/k01ek/netbox-qrcode/issues/36

Roughly, the new feature would "intelligently" determine how to prepopulate some key fields in an 'add device' scenario.
The design would support multi-vendor capabilities, forking off to vendor-specific logic based on incoming information ( a "plugin-within-a-plugin" if you will).

Would appreciate any thoughts, concerns, expressions of support/interest

**Comment by FlorianHeigl** [comment]
absolutely... FYI this idea/issue was really had to find, i have been looking at and trying netbox-qrcode over multiple years and never found this [`netbox-qrprovision`](https://jbemmel.github.io/netbox-qrprovision/) with the demo scanning code.

what example QR codes do you know? or how would you pregenerate them? I'll scan the stuff in my office to see what the vendors put there.

**Comment by SorianoTech** [comment]
Hi! 👋

I'm currently working on a plugin that reads QR codes from Dell devices (at least for now).  
The plugin has two main features:

- Reading/scanning servicetag devices that are **already installed in a rack**.
- Scanning devices when they **arrive at the warehouse** so they can be added to NetBox inventory right away.

This two features contact to dell website to obtain the decive type in case need it.

I'll let you know once it's running in production. 😄
